classes/order/OrderDetail.php

```diff
@@ -520,8 +520,16 @@ protected function setSpecificPrice(Order $order, $product = null)
 					$tax_manager = TaxManagerFactory::getManager($this->vat_address, $id_tax_rules);
 					$this->tax_calculator = $tax_manager->getTaxCalculator();
 
-					$this->reduction_amount_tax_incl = $this->reduction_amount;
-					$this->reduction_amount_tax_excl = Tools::ps_round($this->tax_calculator->removeTaxes($this->reduction_amount_tax_incl), 2);
+					if ($this->reduction_tax)
+					{
+						$this->reduction_amount_tax_incl = $this->reduction_amount;
+						$this->reduction_amount_tax_excl = Tools::ps_round($this->tax_calculator->removeTaxes($this->reduction_amount), 2);
```

Comment:
Shouldn't it be _PS_PRICE_COMPUTE_PRECISION_ instead of the hard-coded 2 in ps_round?